sds_data_manager/lambda_code/IAlirtCode/ialirt_data_query_api.py

```diff
@@ -0,0 +1,249 @@
+"""I-ALiRT Data Query lambda."""
+
+import json
+import logging
+import os
+import time
+from datetime import datetime, timedelta, timezone
+from urllib.parse import quote_plus, unquote_plus, urlencode
+
+import boto3
+from boto3.dynamodb.conditions import Key
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+table_name = os.environ.get("DATA_TABLE")
+region = os.environ.get("AWS_DEFAULT_REGION", "us-west-2")
+dynamodb = boto3.resource("dynamodb", region_name=region)
+table = dynamodb.Table(table_name)
+
+
+def apply_time_filters(params: dict, query_kwargs: dict) -> Key:
+    """Apply the filters for time.
+
+    Parameters
+    ----------
+    params : dict
+        Event parameters.
+    query_kwargs : dict
+        Query keyword arguments.
+
+    Returns
+    -------
+    key_expr : Key
+        The updated key expression with time filters applied.
+    """
+    key_expr = query_kwargs["KeyConditionExpression"]
+
+    start = params.get("time_utc_start") or params.get("met_in_utc_start")
+    end = params.get("time_utc_end") or params.get("met_in_utc_end")
+
+    if start and end:
```

Comment:
Do you want to do any validation on the times? Something like `datetime.strptime(start, "YYYY-mm-ddTHH:MM:SS")` or whatever format you are expecting. Otherwise I'm uncertain whether you could get some weird situations with someone saying time_utc_start="1" to essentially do a full table scan because it would return results. I feel like I am always struggling with datetimes and formatting expectations with APIs, so a nice error message back saying what you expect a user to submit can be really helpful.